dsa-solutions/lc-solutions/0200-0299/0211-Design Add and Search Words Data Structure.md

```diff
@@ -0,0 +1,197 @@
+---
+id: Design Add and Search Words Data Structure
+title: Design Add and Search Words Data Structure
+sidebar_label: 0211-Design Add and Search Words Data Structure
+tags:
+  - Strings
+  - C++
+  - Python
+description: "This document provides a solution to the Design Add and Search Words Data Structure Problem."
+---
+## Problem Statement
+Design a data structure that supports adding new words and finding if a string matches any previously added string.
+
+Implement the WordDictionary class:
+
+- WordDictionary() Initializes the object.
+- void addWord(word) Adds word to the data structure, it can be matched later.
+- bool search(word) Returns true if there is any string in the data structure that matches word or false otherwise. word may contain dots '.' where dots can be matched with any letter.
+
+### Example:
+#### Input:
+
+["WordDictionary","addWord","addWord","addWord","search","search","search","search"]
+[[],["bad"],["dad"],["mad"],["pad"],["bad"],[".ad"],["b.."]]
+
+#### Output:
+[null,null,null,null,false,true,true,true]
+
```

Comment:
replace

```md
#### Explanation:
- WordDictionary wordDictionary = new WordDictionary();
- wordDictionary.addWord("bad");
- wordDictionary.addWord("dad");
- wordDictionary.addWord("mad");
- wordDictionary.search("pad"); // return False
- wordDictionary.search("bad"); // return True
- wordDictionary.search(".ad"); // return True
- wordDictionary.search("b.."); // return True
```

to

````md
#### Explanation:
```
WordDictionary wordDictionary = new WordDictionary();
wordDictionary.addWord("bad");
wordDictionary.addWord("dad");
wordDictionary.addWord("mad");
wordDictionary.search("pad"); // return False
wordDictionary.search("bad"); // return True
wordDictionary.search(".ad"); // return True
wordDictionary.search("b.."); // return True
```
````